sympy/solvers/tests/test_ode.py

```diff
@@ -3332,22 +3320,6 @@ def test_nth_algebraic_redundant_solutions():
     assert sol == dsolve(eqn, f(x))
 
 
-#
-# These tests can be combined with the above test if they get fixed
-# so that dsolve actually works in all these cases.
-#
-
-# Fails due to division by f(x) eliminating the solution before nth_algebraic
-# is called.
-@XFAIL
-def test_nth_algebraic_find_multiple1():
-    eqn = f(x) + f(x)*f(x).diff(x)
-    solns = [Eq(f(x), 0),
-             Eq(f(x), C1 - x)]
-    assert all(c[0] for c in checkodesol(eqn, solns, order=1, solve_for_func=False))
-    assert set(solns) == set(dsolve(eqn, f(x)))
-
-
 # prep = True breaks this
```

Comment:
XFAIL test passes so I have added this into the above test case.

Reply:
You've moved this above but removed the algebraic solution. It is XFAIL here because dsolve does not find the algebraic solution. I don't quite understand what you are doing here.